extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/spi/datafetcher/QuarkusDefaultDataFetcher.java

```diff
@@ -151,4 +151,11 @@ private void deactivate(ManagedContext requestContext) {
         SmallRyeContextManager.clearCurrentSmallRyeContext();
         requestContext.deactivate();
     }
+
+    private Context getContext() {
+        Context vc = Vertx.currentContext();
+        if (vc != null)
+            return vc;
+        return Vertx.vertx().getOrCreateContext();
```

Comment:
Hold on, this creates a new vertx context that we don't close anywhere, no? And it would happen again and again. Are you sure this is not a massive resource leak?

Reply:
@cescoffier should probably look at this

Reply:
Yes, we cannot do that, that creates a new Vert.x instance on every call and .... these instances will never be closed.

Reply:
What do you mean be closed ? I.e how do I close an instance ?

Reply:
We shouldn't create any new context instances, I think we need to figure out why `QuarkusDefaultDataFetcher.invokeAndTransform` is sometimes called on a ForkJoinPool thread instead of a Vert.x worker thread. It seems to be random, and when executed in the ForkJoinPool, it fails

Reply:
You can either call
```
CompletableFuture.supplyAsync(supplier, Infrastucture.getDefaultWorkerPool())
```
or obtain the worker pool by `@Inject`ing a `java.util.concurrent.Executor` and pass that one there. I think Quarkus can't do anything about applications spawning tasks in the wrong thread pools.

Reply:
In any case, we should probably provide a better error message and avoid throwing a NPE :).